v.30: Support gzip, brotli, and deflate (#1987)
meilisearch/documentation: reference/api/overview.md

Only the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints accept NDJSON and CSV. For all others, use `Content-Type: application/json`.

### Content encoding

> The `Content-Encoding` header compresses the media type. It improves transfer speed and reduces bandwidth consumption by sending and receiving a smaller, compressed payload. The `Accept-Encoding` header indicates the compression algorithm the client understands. Meilisearch supports the following compression methods:

Reviewer: ```suggestion
The `Content-Encoding` header indicates that the media type is compressed by the given algorithm. It improves transfer speed and reduces bandwidth consumption by sending and receiving a smaller, compressed payload. The `Accept-Encoding` header indicates the compression algorithm the client understands. Meilisearch supports the following compression methods:
```

> The `Content-Encoding` header indicates that the media type is compressed by a given algorithm. It improves transfer speed and reduces bandwidth consumption by sending and receiving a smaller, compressed payload. The `Accept-Encoding` header indicates the compression algorithm the client understands. Meilisearch supports the following compression methods:

Reviewer 2: Since `content-encoding` and `accept-encoding` are so similar, I feel the "instead" helps reinforcing they're different things.

```suggestion
The `Content-Encoding` header indicates the media type is compressed by a given algorithm. Compression improves transfer speed and reduces bandwidth consumption by sending and receiving smaller payloads. The `Accept-Encoding` header,  instead, indicates the compression algorithm the client understands. 

Meilisearch supports the following compression methods:
```

> The `Content-Encoding` header indicates that the media type is compressed by the given algorithm. It improves transfer speed and reduces bandwidth consumption by sending and receiving a smaller, compressed payload. The `Accept-Encoding` header indicates the compression algorithm the client understands. Meilisearch supports the following compression methods:

Author: ```suggestion
The `Content-Encoding` header indicates that the media type is compressed by a given algorithm. It improves transfer speed and reduces bandwidth consumption by sending and receiving a smaller, compressed payload. The `Accept-Encoding` header indicates the compression algorithm the client understands. Meilisearch supports the following compression methods:
```

- `br`: uses the [Brotli](https://en.wikipedia.org/wiki/Brotli) algorithm
- `deflate`: uses the [zlib](https://en.wikipedia.org/wiki/Zlib) structure with the [deflate](https://en.wikipedia.org/wiki/DEFLATE) compression algorithm

> - `gzip`: uses the [Lempel-Ziv](https://en.wikipedia.org/wiki/LZ77_and_LZ78#LZ77) coding (LZ77) with a 32-bit CRC

Reviewer: ```suggestion
- `gzip`: uses the [GZip](https://en.wikipedia.org/wiki/Gzip) algorithm
```

#### Request compression

> The code sample below compresses the request using the `Content-Encoding: gzip` header:

Reviewer: ```suggestion
The code sample below indicates that the request body is compressed by the given algorithm using the `Content-Encoding: gzip` header:
```

> The code sample below compresses the request using the `Content-Encoding: gzip` header:

Author: ```suggestion
The code sample below uses the `Content-Encoding: gzip` header, indicating that the request body is compressed using the `gzip` algorithm:
```

<CodeSamples id="api_reference_request_gzip_1" />

#### Response compression

> Meilisearch compresses a response if the request contains the `Accept-Encoding: gzip` header:

Author: ```suggestion
Meilisearch compresses a response if the request contains the `Accept-Encoding` header. The code sample below uses the `gzip` algorithm:
```

Only the [add documents](/reference/api/documents.md#add-or-replace-documents) and [update documents](/reference/api/documents.md#add-or-update-documents) endpoints accept NDJSON and CSV. For all others, use `Content-Type: application/json`.

### Content encoding

> The `Content-Encoding` header indicates the media type is compressed by a given algorithm. Compression improves transfer speed and reduces bandwidth consumption by sending and receiving smaller payloads. The `Accept-Encoding` header,  instead, indicates the compression algorithm the client understands. 

Author: ```suggestion
The `Content-Encoding` header indicates the media type is compressed by a given algorithm. Compression improves transfer speed and reduces bandwidth consumption by sending and receiving smaller payloads. The `Accept-Encoding` header, instead, indicates the compression algorithm the client understands. 
```
